lib/fathead/ruby/parse.rb

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Holds database column values to be output as tab-separated values.
+class OutputRow
+  HEADERS = [
+    :title,
+    :type,
+    :redirect_title,
+    nil,
+    :categories,
+    nil,
+    :related,
+    nil,
+    :links,
+    :disambiguation_content,
+    :image,
+    :abstract,
+    :url
+  ].freeze
+
+  attr_accessor(*HEADERS.compact)
+
+  def initialize
+    yield self
+    freeze
+  end
+
+  def to_s
+    HEADERS.map { |header| escape public_send header if header } * "\t" << "\n"
+  end
+
+  private
+
+  def escape(text)
+    String(text).gsub(/[\\\n\t]/, '\\' => '\\\\', "\n" => '\n', "\t" => '\t')
+  end
+end
+
+# Abstract description of documentation.
+class Documentation
+  attr_accessor :title, :url, :description, :categories, :related, :alt_titles
+
+  def initialize
+    yield self
+    freeze
+  end
+
+  def abstract
+    intro.chunk(&:name).map do |name, elements|
+      if name == 'pre'
+        elements.map { |e| code_block e.text }.join
+      else
+        elements.map { |e| body_text e.text }.join('<br>')
+      end
+    end.unshift(usage).join
+  end
+
+  def to_s
+    [to_row, to_redirect_rows].join
+  end
+
+  def breaks_category_page?
+    title.include?('[') || abstract.include?('[')
+  end
+
+  private
+
+  def to_row
+    OutputRow.new do |row|
+      row.title = title
+      row.type = 'A' # article
+      row.categories = categories.join("\n")
```

Comment:
@kotoshenya looking at the output.txt though, I'm seeing the categories are joined with `\n` (instead of `\\n`) so the result is that only the last category is being recognized by our parser internally.

This causes the infobox to only link to one category instead of multiple.

e.g. https://duckduckgo.com/?q=ruby+string+hex&ia=about

Should be linking to two categories: `Ruby instance methods` and `Ruby String methods`